config/initializers/active_record.rb

```diff
@@ -11,7 +11,8 @@ def with_statement_timeout(seconds)
       execute("SET statement_timeout = '#{seconds * 1000}';")
       yield(self)
     ensure
-      execute("SET statement_timeout = '#{ActiveRecord::Base.configurations.dig(Rails.env, 'variables', 'statement_timeout')}';")
+      default_timeout = ActiveRecord::Base.configurations.find_db_config(Rails.env).config.dig("variables", "statement_timeout")
```

Comment:
looks good, but just a headsup that `config` will be gone in Rails 7.0:

`DEPRECATION WARNING: DatabaseConfig#config will be removed in 7.0.0 in favor of DatabaseConfig#configuration_hash which returns a hash with symbol keys (called from <main> at bin/rails:6)`